### ISSUE TITLE
Allele_Frequency column is emitted empty 

### ISSUE THREAD

**Issue by biotechnepal** [ambj/MuPeXI#28]
Hi,
I wonder why "Allele_Frequency" filed is empty i.e "-" in *.mupexi output.
I am using vcf from GATK4-Mutect2.

Regards,
Raj
--

**Comment by biotechnepal** [comment]
#A notice should be taken if TUMOR sample is in col. 9 or 10 of vcf file. This script assumes that it's in column 9, but by Mutect2, it can be in col 10.
#"Mutect2"  should be added in line 497.

**Comment by kobejamescurry** [comment]
> R sample is in
The example.vcf  seems to in column 10, run is ok
![image](https://user-images.githubusercontent.com/46668881/51620249-f0e55080-1f6c-11e9-851e-5f24f7326251.png)

**Comment by kobejamescurry** [comment]
> #A notice should be taken if TUMOR sample is in col. 9 or 10 of vcf file. This script assumes that it's in column 9, but by Mutect2, it can be in col 10.
> #"Mutect2" should be added in line 497.

do you have some ideas about how to cut peptides including mutant base. through transcript or cds?

**Comment by kobejamescurry** [comment]
do you know how the result of column driver gene comes out.
I see in cosmic there is no clear definition of driver gene

**Comment by ibwoo** [comment]
> Hi,
> I wonder why "Allele_Frequency" filed is empty i.e "-" in *.mupexi output.
> I am using vcf from GATK4-Mutect2.
> 
> ## Regards,
> Raj

Hi @biotechnepal,
If you're using GATK4, [this issue has already been resolved and @ambj has pushed the fix to the 'master-branch', but perhaps this is not yet in the release version.](https://github.com/ambj/MuPeXI/issues/12#L417)

A quick fix would be to check the VCF file header line #ID and change this to reflect the old MuTect2 case-sensitive naming. MuPeXI is looking for "ID=MuTect2" and GATK4 supplies "ID=Mutect2", if I'm not mistaken. 

Once this is recognised correctly by MuPeXI, the allele frequencies should be correctly taken from your vcf file if everything else is in order. 

I hope that's helpful.

**Comment by kobejamescurry** [comment]
@ibwoo  I have test with gatk4.0.11.0, it is ok, but do you know how ambj define driver gene, in cosmic vcf, there are only TSG, oncogene, and another(I forget), how she do that, thanks a lot

**Comment by ibwoo** [comment]
@kobejamescurry this sounds like it should be a separate issue? Are you
talking about the Yes/No column in MuPeXI output? If so, MuPeXI just checks
each mutant against (I think) the "symbol" column of the cancer concensus
file. It's basically just saying is this mutation in a gene that's
associated with cancer. You could substitute that cosmic file with any
other gene list. In my case, I believe this file was a csv so if you have a
vcf you may have downloaded the wrong file or maybe I've misunderstood your
question.

**Comment by kobejamescurry** [comment]
@ibwoo Thanks for your continuous help and attention.
here is the link ambj supplied https://cancer.sanger.ac.uk/census

**Comment by ibwoo** [comment]
@kobejamescurry in that link you just sent me, I believe that if you click the 'csv' button and sign in or make an account, I think that's where you get the list in CSV format. Again, I'm not sure where you found the "cosmic vcf" but perhaps you mis-typed.

**Comment by kobejamescurry** [comment]
@ibwoo yes, you are right, here need is not vcf, but tsv, just sign up and save the tsv link, you will see that, do you have a skype account, I think we can have a better communication if you would like to, thanks a lot

**Comment by ibwoo** [comment]
@kobejamescurry It's not clear what your question is, if it was not answered by [my earlier response](https://github.com/ambj/MuPeXI/issues/28#issuecomment-457217330). This 'Cancer Gene Census' list on Cosmic is used by MuPeXI as part of their additional 'Annotation' information. I've explained how the MuPeXI program accesses the downloaded .csv and checks the gene symbols against that list. It then adds something like "Yes" or "TRUE" (I can't remember off the top of my head) for a positive hit in the 'Cancer Driver Gene" column of MuPeXI output. 

This column is only used as extra annotation to help the end user, and so you can choose to filter that Cosmic gene list, supply your own or omit it entirely and it won't matter. If you are concerned that you want to have the exact same version of the list that the MuPeXI webserver uses, you would have to ask Anne-Mette, as Cosmic constantly updates their list (last updated in 2019) and it's unlikely that the MuPeXI webserver keeps it up to date.

**Comment by kobejamescurry** [comment]
@ibwoo thanks a lot, thanks for your contunious attention.
you can just click the link supplied by ambj ### **http://cancer.sanger.ac.uk/census**

**Comment by ShahiRB** [comment]
> > Hi,
> > I wonder why "Allele_Frequency" filed is empty i.e "-" in *.mupexi output.
> > I am using vcf from GATK4-Mutect2.
> > ## Regards,
> > Raj
> 
> Hi @biotechnepal,
> If you're using GATK4, [this issue has already been resolved and @ambj has pushed the fix to the 'master-branch', but perhaps this is not yet in the release version.](https://github.com/ambj/MuPeXI/issues/12#L417)
> 
> A quick fix would be to check the VCF file header line #ID and change this to reflect the old MuTect2 case-sensitive naming. MuPeXI is looking for "ID=MuTect2" and GATK4 supplies "ID=Mutect2", if I'm not mistaken.
> 
> Once this is recognised correctly by MuPeXI, the allele frequencies should be correctly taken from your vcf file if everything else is in order.
> 
> I hope that's helpful.

Hi @ibwoo , Thanx!
